source/PlayerInfo.cpp

```diff
@@ -2698,11 +2771,131 @@ void PlayerInfo::CreateMissions()
 		}
 	}
 
-	// Sort missions on the job board alphabetically.
-	availableJobs.sort([](const Mission &lhs, const Mission &rhs)
+	// Sort missions on the job board.
+	SortAvailable();
+}
+
+
+
+void PlayerInfo::SortAvailable()
+{
+	// Destinations: planets OR system. Only counting them, so the type doesn't matter.
+	set<const void*> destinations;
+	if(availableSortType == CONVENIENT)
+	{
+		for(const Mission &mission : Missions())
+		{
+			if(mission.IsVisible())
+			{
+				destinations.insert(mission.Destination());
+				destinations.insert(mission.Destination()->GetSystem());
+
+				for(const Planet* stopover : mission.Stopovers())
+				{
+					destinations.insert(stopover);
+					destinations.insert(stopover->GetSystem());
+				}
+
+				for(const System* waypoint : mission.Waypoints())
+					destinations.insert(waypoint);
+			}
+		}
+	}
+	availableJobs.sort([&](const Mission &lhs, const Mission &rhs)
 	{
-		return lhs.Name() < rhs.Name();
+		// First, separate rush orders if wanted
+		if(sortSeparateRush)
+		{
+			if(!lhs.Deadline() && rhs.Deadline())
+				return availableSortAsc; //availableSortAsc instead of true, to counter the actual reversal below
+			if(lhs.Deadline() && !rhs.Deadline())
+				return !availableSortAsc;
+		}
+		// Then, separate greyed-out jobs you can't accept
+		if(sortSeparateGray)
```

Comment:
sortSeparateGray doesn't seem very clear either, the color we attribute to jobs that can't be taken shouldn't be used to name this
How about sortPossibleMissions?